articles/text-to-speech-in-javascript/index.md

```diff
@@ -0,0 +1,391 @@
+      <div class="mx-5">
+        <p class="lead">Rate</p>
+        <input type="range" min="0.1" max="10" value="1" id="rate" step="0.1" />
+        <span id="rate-label" class="ms-2">1</span>
+      </div>
+      <div>
+        <p class="lead">Pitch</p>
+        <input type="range" min="0" max="2" value="1" step="0.1" id="pitch" />
+        <span id="pitch-label" class="ms-2">1</span>
+      </div>
+    </div>
+
+    <!-- Text Area  for the User to Type -->
+    <textarea class="form-control bg-dark text-light mt-5" cols="30" rows="10" placeholder="Type here..."></textarea>
+
+    <!-- Control Buttons -->
+    <div class="mb-5">
+      <button id="start" class="btn btn-success mt-5 me-3">Start</button>
+      <button id="pause" class="btn btn-warning mt-5 me-3">Pause</button>
+      <button id="resume" class="btn btn-info mt-5 me-3">Resume</button>
+      <button id="cancel" class="btn btn-danger mt-5 me-3">Cancel</button>
+    </div>
+  </body>
+  <script src="./textToSpeech.js"></script>
+</html>
+```
+
+![HTML Web Page](start_screen.png)
+
+### The JavaScript file
+Let's create an instance of the [`SpeechSynthesisUtterance`](https://developer.mozilla.org/en-US/docs/Web/API/SpeechSynthesisUtterance) class. We'll configure this instance with various properties.
+
+```JavaScript
+let speech = new SpeechSynthesisUtterance();
+```
+
+#### Properties
+Now, let's configure some properties on this `SpeechSynthesisUtterance` instance.
+
+There are 6 properties on the `SpeechSynthesisUtterance` instance that we can tweak. They are:
+
+**Language**:
+
+The `language` property gets and sets the language of the utterance. If unset, the `<html lang="en">` lang value will be used, or the user-agent default if the `<html lang="en">` lang is unset.
+
+It accepts a DOMString representing a [BCP 47 language tag](https://en.wikipedia.org/wiki/IETF_language_tag).
+
+```JavaScript
+speech.lang = "en";
+```
+
+**Text**:
+
+The `text` property gets and sets the text that will be synthesized when the utterance is spoken. The text can be provided as plain text. In our case, the text property must be set when the start button is clicked.
+
+Let's add a click listener to the button. When the button is clicked, we should get the text value from the `textarea` and set it to this property.
+
+You can learn more about event listeners [here](https://www.w3schools.com/js/js_htmldom_eventlistener.asp).
+
+```JavaScript
+document.querySelector("#talk").addEventListener("click", () => {
+  speech.text = document.querySelector("textarea").value;
+});
+```
+
+**Volume**:
+
+The `volume` property gets and sets the volume of the utterance. It is a float that represents the volume value, between 0 (lowest) and 1 (highest). The default value is 1 if this property is unset.
+
+Let's add a `onInput` listener to the `volume` range slider and adjust the `volume` property when the value of the slider changes. We've already set the min, max and default value of the slider in the HTML tag.
```

Comment:
> a `onInput` listener

Wouldn't "an `onInput` listener" be more appropriate?

Reply:
yep. not sure how I missed it 😅